doc/source/server_architecture.rst

```diff
@@ -61,21 +61,29 @@ AIT provides a number of default plugins. Check the `Plugins API documentation <
+- **Inbound streams** can have an address specification or inbound streams as their **input**. Inbound streams can have multiple inputs.
 
     - The server sets up an input stream that emits properly formed telemetry packet messages over a globally configured topic. This is used internally by the ground script API for telemetry monitoring. The input streams that pass data to this stream must output data in the Packet UID annotated format that the core packet handlers use. The input streams used can be configured via the **server.api-telemetry-streams** field. If no configuration is provided the server will default to all valid input streams if possible. See :ref:`the Ground Script API documentation <api_telem_setup>` for additional information.
 
 - **Outbound streams** can have plugins or outbound streams as their **input**. Outbound streams can have multiple inputs.
 
-   - Outbound streams also have the option to **output** to an integer port (see :ref:`example config below <Stream_config>`).
+   - Outbound streams also have the option to **output** to an address specification (see :ref:`example config below <Stream_config>`).
 
    - The server exposes an entry point for commands submitted by other processes. During initialization, this entry point will be connected to a single outbound stream, either explicitly declared by the stream (by setting the **command-subscriber** field; see :ref:`example config below <Stream_config>`), or decided by the server (select the first outbound stream in the configuration file).
 
 - Streams can have any number of **handlers**. A stream passes each received *packet* through its handlers in order and publishes the result.
-- There are several stream classes that inherit from the base stream class. These child classes exist for handling the input and output of streams differently based on whether the inputs/output are ports or other streams and plugins. The appropriate stream type will be instantiated based on whether the stream is an inbound or outbound stream and based on the inputs/output specified in the stream's configs. If the input type of an inbound stream is an integer, it will be assumed to be a port. If it is a string, it will be assumed to be another stream name or plugin. Only outbound streams can have an output, and the output must be a port, not another stream or plugin.
+- There are several stream classes that inherit from the base stream class. These child classes exist for handling the input and output of streams differently based on whether the inputs/output are remote hosts, ports or other streams and plugins. The appropriate stream type will be instantiated based on whether the stream is an inbound or outbound stream and based on the inputs/output specified in the stream's configs. Only outbound streams can have an output, and the output must be an address specification, not another stream or plugin.
 
 .. _Stream_config:
 
+TCP/UDP Address Specification:
+
```

Comment:
Per the doc updates above, could a note be added that states, for backward compatibility, integer ports are treated as 'udp:localhost:port'